File: src/main/java/com/sleepkqq/sololeveling/view/auth/TelegramAuthComponent.java
package com.sleepkqq.sololeveling.view.auth;

import com.sleepkqq.sololeveling.service.auth.TgAuthService;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.react.ReactAdapterComponent;

@Tag("telegram-auth")
@JsModule("./component/TelegramAuthComponent.js")
public class TelegramAuthComponent extends ReactAdapterComponent {

  private static final String TG_AUTH_DATA = "tgAuthData";

  private final TgAuthService tgAuthService;

  public TelegramAuthComponent(TgAuthService tgAuthService) {
    this.tgAuthService = tgAuthService;
    addAuthListener();
  }

  private void addAuthListener() {
    addStateChangeListener(TG_AUTH_DATA, TgAuthData.class, tgAuthService::authenticate);
  }
}

File: solo-leveling-ui-service/src/main/kotlin/com.sleepkqq.sololeveling/ui/view/auth/TelegramAuthComponent.kt
package com.sleepkqq.sololeveling.ui.view.auth

import com.sleepkqq.sololeveling.ui.model.TgAuthData
import com.sleepkqq.sololeveling.ui.service.TgAuthService
import com.vaadin.flow.component.Tag
import com.vaadin.flow.component.dependency.JsModule
import com.vaadin.flow.component.react.ReactAdapterComponent

@Tag("telegram-auth")
@JsModule("./component/TelegramAuthComponent.js")
class TelegramAuthComponent(
	private val tgAuthService: TgAuthService
) : ReactAdapterComponent() {

	companion object {

		private const val TG_AUTH_DATA = "tgAuthData"
	}

	init {
		addAuthListener()
	}

	private fun addAuthListener() =
		addStateChangeListener(TG_AUTH_DATA, TgAuthData::class.java, tgAuthService::authenticate)
}